### PR TITLE
remove UID mutation from request.context

### LINKED ISSUE #59775: Consider removing UID mutation from request.context
Injecting a UID into the context of a request via `WithUID` will cause it to override the UID of an object (nothing does this currently):

https://github.com/kubernetes/kubernetes/blob/d4eaf0b6801900d2b2f80036e267a1d3d86fd3a8/staging/src/k8s.io/apiserver/pkg/registry/rest/meta.go#L28-L30

It is unclear when this would be useful or safe.  There is also concern in regards to `MutatingWebhooks` that set a UID on the object (`FillObjectMetaSystemFields` prevents this from being an issue):

https://groups.google.com/d/msg/kubernetes-sig-architecture/mVGobfD4TpY/vIJXeQTMAQAJ

Perhaps we should remove `WithUID`?

@kubernetes/sig-api-machinery-misc

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**What this PR does / why we need it**:

remove UID mutation from request.context, which is no use currently.

Fixes #59775

**Special notes for your reviewer**:

**Release note**:

```release-note
Remove UID mutation from request.context.
```

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
cc @enj @cheftako

**Comment by hzxuzhonghu** [comment]
cc @lavalamp @liggitt

File: staging/src/k8s.io/apiserver/pkg/registry/rest/meta.go [hunk 1]
```diff
@@ -25,15 +25,9 @@ import (
 )
 
 // FillObjectMetaSystemFields populates fields that are managed by the system on ObjectMeta.
-func FillObjectMetaSystemFields(ctx context.Context, meta metav1.Object) {
+func FillObjectMetaSystemFields(meta metav1.Object) {
 	meta.SetCreationTimestamp(metav1.Now())
-	// allows admission controllers to assign a UID earlier in the request processing
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/registry/rest/meta.go]: @liggitt @deads2k do we use this in origin?

Comment (sttts) [reply staging/src/k8s.io/apiserver/pkg/registry/rest/meta.go]: don't seeing a call there at least to WithUID

Comment (sttts) [reply staging/src/k8s.io/apiserver/pkg/registry/rest/meta.go]: ok, https://github.com/kubernetes/kubernetes/issues/59775#issuecomment-389735867

**Pull request merged**